core/coreapi/services/netparams.go

```diff
@@ -49,13 +50,28 @@ func NewNetParams(ctx context.Context) (netParams *NetParams) {
 }
 
 func (a *NetParams) consume() {
-	defer func() { close(a.ch) }()
+	defer func() {
+		// clear the channel before we close it
+		// can't use a WaitGroup as suggested because if we get to this point
+		// if the Push is still in progress and the queue is full, we'll
+		// end up in a situation where we're waiting for the queue to empty, but
+		// no consumer.
+		for range a.ch {
```

Comment:
Question, could you not use buffered channel instead?

Reply:
I'm pretty sure the channel is already buffered.